go.mod

```diff
@@ -89,7 +89,7 @@ require (
 	gonum.org/v1/gonum v0.6.2
 	google.golang.org/api v0.60.0
 	google.golang.org/genproto v0.0.0-20220502173005-c8bf987b8c21
-	google.golang.org/grpc v1.47.0
+	google.golang.org/grpc v1.49.0
```

Comment:
curious what prompted this change, etcd 3.5.5 is still on 1.41.0 (https://github.com/etcd-io/etcd/blob/release-3.5/go.mod#L35)

nothing breaking-looking jumps out at me from https://github.com/grpc/grpc-go/releases, but the diff isn't very reviewable, so just wondering

Reply:
i did not do this explicitly, it may have some from one of the otel libraries i think.

Reply:
Mm, ok.

Reply:
if i force to pin the dependency on the old version (`1.47.0`) here's what happens.

```
[dims@dims-m1-7728 17:56] ~/go/src/k8s.io/kubernetes ⟩ hack/pin-dependency.sh google.golang.org/grpc v1.47.0
Running: go mod download google.golang.org/grpc@v1.47.0
Resolved to google.golang.org/grpc@v1.47.0
Running: go mod edit -require google.golang.org/grpc@v1.47.0
Running: go mod edit -replace google.golang.org/grpc=google.golang.org/grpc@v1.47.0

Run hack/update-vendor.sh to rebuild the vendor directory


[dims@dims-m1-7728 17:56] ~/go/src/k8s.io/kubernetes ⟩ hack/update-vendor.sh
+++ [0919 17:56:49] logfile at /tmp/update-vendor.Py7b/update-vendor.log
+++ [0919 17:56:49] go.mod: update staging references
+++ [0919 17:56:57] go.mod: propagate to staging modules
+++ [0919 17:56:57] go.mod: sorting staging modules
+++ [0919 17:57:12] go.mod: tidying
+++ [0919 17:57:23] go.mod: adding generated comments
+++ [0919 17:57:23] vendor: updating internal modules
+++ [0919 17:57:23] vendor: running 'go mod vendor'
+++ [0919 17:57:27] vendor: updating vendor/LICENSES
+++ [0919 17:57:33] vendor: creating OWNERS file
+++ [0919 17:57:33] NOTE: don't forget to handle vendor/* files that were added or removed


[dims@dims-m1-7728 17:57] ~/go/src/k8s.io/kubernetes ⟩ hack/lint-dependencies.sh
Check all unwanted dependencies and update its status.
These modules are pinned to versions different than the minimal preferred version.
That means that without replace directives, a different version would be selected,
which breaks consumers of our published modules.
1. Use hack/pin-dependency.sh to switch to the preferred version for each module
2. Run hack/update-vendor.sh to rebuild the vendor directory
3. Run hack/lint-dependencies.sh to verify no additional changes are required

google.golang.org/grpc
    pinned:    v1.47.0
    preferred: v1.49.0
    hack/pin-dependency.sh google.golang.org/grpc v1.49.0
All pinned versions of checked dependencies match their preferred version.
```

Reply:
it's coming from here:
https://github.com/open-telemetry/opentelemetry-go-contrib/blob/instrumentation/google.golang.org/grpc/otelgrpc/v0.35.0/instrumentation/google.golang.org/grpc/otelgrpc/go.mod#L9

`go mod graph` shows:
```
go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc@v0.35.0 google.golang.org/grpc@v1.49.0
```